cartographer_ros_msgs/msg/StatusResponse.msg

```diff
@@ -0,0 +1,17 @@
+# Copyright 2018 The Cartographer Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A common message type to indicate the outcome of a service call.
+uint8 code
```

Comment:
Maybe this should have been a `StatusCode` (and have `uint8 code` inside there)? Personally I would have preferred to have everything inside `StatusResponse`, since `StatusCode` as a message type isn't used anywhere.

Reply:
If you mean to add a variable `uint8 code` to `StatusCode.msg` and write `StatusResponse.msg` like that:
```
cartographer_ros_msgs/StatusCode code
string message
```
...I tried, but this doesn't compile unfortunately. One probably has to live with the fact that ROS msgs don't support anything close to a real strongly typed enum 😒 (in real languages you would ensure that the code can't be out of the bounds of the gRPC codes, i.e. not "plain" uint8)

Regarding your second point: it _is_ used, however not as a stand-alone message but more as a constant definition in msg format that can be used internally (see e.g. [here](https://github.com/googlecartographer/cartographer_ros/blob/512a9cc35837f1ac3738edadd908b88a0843a0de/cartographer_ros/cartographer_ros/node.cc#L413)) or by service callers, e.g. a Python node. Having everything declared in a single msg might be possible, but I'm afraid it would change my example code to sth. like this:
```cpp
switch (response.status.code) {
  case cartographer_ros_msgs::StatusResponse::OK: return true;  // before: cartographer_ros_msgs::StatusCode::OK
  case cartographer_ros_msgs::StatusResponse::CANCELLED: return false; // before: cartographer_ros_msgs::StatusCode::CANCELLED
  // ...
}
```
which is not as consistent with the naming as before.

Reply:
The way you wrote that switch looks good to me (even semantically), I would have preferred it like that :-)